### ISSUE TITLE
consistent-this incorrectly triggered when destructuring `this`

### ISSUE THREAD

**Issue by davidaurelio** [eslint/eslint#2633]
```js
const {a, b, c} = this;
```

Triggers `consistent-this`, but it should not.
